### DISCUSSION TITLE
Is there a way to invalidate useFirestoreDocDataOnce when history.push happened

### DISCUSSION THREAD

**Discussion by banyan** [FirebaseExtended/reactfire#259] [Q&A] [answered]
Transition to a page with `history.push` to get the required document using `useFirestoreDocDataOnce`. In the meantime, no realtime updates are required. There was an update in the first document when I went to a different page. Go back to the original page again, here I want to run `useFirestoreDocDataOnce` again, is there a way to do that?

subscribe and unsubscribe in `useEffect` might be worked but the hook can not put there. 

https://codesandbox.io/s/reactfire-usefirestoredocdataonce-05i51?file=/src/config.js

![2020-05-22 00 50 58](https://user-images.githubusercontent.com/19625/82577739-5b8cb780-9bc6-11ea-98a5-518303af2027.gif)

1. counter is 2
2. increment by 1
3. `useFirestoreDocDataOnce` keeps 2

**Comment by jamesdaniels** [comment, marked as answer]
The realtime snapshot listener will timeout in 30 seconds, after it is no longer in the view-tree.
